### PR TITLE
Adds the Interchain Queries Spec (ICS-032)

### PR DESCRIPTION
**Pull request by boojamya**
Adds spec info for Interchain Queries candidate (ICS-032).

It has been decided to house and maintain this implementation on [strangelove-ventures/async-icq](https://github.com/strangelove-ventures/async-icq) repo.

### CODE REVIEW

File: spec/app/ics-032-interchain-queries/README.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+### ABCI Query
+
+ABCI RequestQuery enables blockchains to request information made by the end-users of applications. A query is received by a full node through its consensus engine and relayed to the application via the ABCI. It is then routed to the appropriate module via BaseApp's query router so that it can be processed by the module's query service
+
+ICQ can only return information from stale reads, for a read that requires consensus, ICA (ICS-27) will be used.
+
+
+#### **SendQuery**
+
+`SendQuery` is used to send an IBC packet containing query information to an interchain account on a host chain.
```
Comment (akure) [inline spec/app/ics-032-interchain-queries/README.md:49]: interchain account -> ICQ module on the host chain.

File: spec/app/ics-032-interchain-queries/README.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+This standard document specifies chain structure, packet flow and authentication logic for the Interchain Queries module. This allows for one blockchain to query for on-chain data from another block chain via ABCI queries.
+
+### Motivation
+
+Interchain Queries enable blockchains to query the state of an account on another chain without the need for ICA auth. ICS-27 Interchain Accounts is used for IBC transactions, e.g. to transfer coins from one interchain account to another, whereas Interchain Queries is used for IBC Query, e.g. to query the balance of an account on another chain. 
+In short, ICA is cross chain writes while ICQ is cross chain reads.
+
+### Definitions 
+
+- `Host Chain`: The chain where the query is sent. The host chain listens for IBC packets from a controller chain that contain instructions (e.g. cosmos SDK messages) that the interchain account will execute.
```
Comment (akure) [inline spec/app/ics-032-interchain-queries/README.md:24]: "controller chain" should be "querying chain"

File: spec/app/ics-032-interchain-queries/README.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+- `Querier Chain`: The chain sending the query to the host chain. The controller chain sends IBC packets to the host chain to query information.
```
Comment (akure) [inline spec/app/ics-032-interchain-queries/README.md:25]: "controller chain" should be "querying chain"

File: spec/app/ics-032-interchain-queries/README.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+- `Interchain Query`: An IBC packet that contains information about the query in the form of ABCI RequestQuery
+
+The chain which sends the query becomes the controller chain, and the chain which receives the query and responds becomes the host chain for the scenario.
```
Comment (akure) [inline spec/app/ics-032-interchain-queries/README.md:28]: "controller chain" should be "querying chain"

File: spec/app/ics-032-interchain-queries/README.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+## Technical Specification
+
+![model](icq-img.png)
+
+
+### ABCI Query
+
+ABCI RequestQuery enables blockchains to request information made by the end-users of applications. A query is received by a full node through its consensus engine and relayed to the application via the ABCI. It is then routed to the appropriate module via BaseApp's query router so that it can be processed by the module's query service
+
+ICQ can only return information from stale reads, for a read that requires consensus, ICA (ICS-27) will be used.
```
Comment (akure) [inline spec/app/ics-032-interchain-queries/README.md:44]: I think, we should remove this. ICA should not be used for query.

Comment (Ehsan-saradar) [reply spec/app/ics-032-interchain-queries/README.md:44]: And I did not get what you mean by "ICQ can only return information from stale reads" 
Maybe its better to remove this line

File: spec/app/ics-032-interchain-queries/README.md (hunk 1 above)
Comment (boojamya) [reply spec/app/ics-032-interchain-queries/README.md:49]: Resolved here: https://github.com/strangelove-ventures/async-icq/pull/8
Moving this readme out of `cosmos/ibc` and into `strangelove-ventures/async-icq`

File: spec/app/ics-032-interchain-queries/README.md (hunk 2 above)
Comment (boojamya) [reply spec/app/ics-032-interchain-queries/README.md:24]: Resolved here: https://github.com/strangelove-ventures/async-icq/pull/8
Moving this readme out of `cosmos/ibc` and into `strangelove-ventures/async-icq`

File: spec/app/ics-032-interchain-queries/README.md (hunk 4 above)
Comment (boojamya) [reply spec/app/ics-032-interchain-queries/README.md:28]: Resolved here: https://github.com/strangelove-ventures/async-icq/pull/8
Moving this readme out of `cosmos/ibc` and into `strangelove-ventures/async-icq`

File: spec/app/ics-032-interchain-queries/README.md (hunk 5 above)
Comment (boojamya) [reply spec/app/ics-032-interchain-queries/README.md:44]: Resolved here: https://github.com/strangelove-ventures/async-icq/pull/8
Moving this readme out of `cosmos/ibc` and into `strangelove-ventures/async-icq`

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**